x-pack/plugin/esql/src/internalClusterTest/java/org/elasticsearch/xpack/esql/plugin/MatchOperatorPushDownIT.java

```diff
@@ -0,0 +1,231 @@
+            // values
+            List<List<Object>> values = getValuesList(resp);
+            assertThat(values.size(), equalTo(1));
+            assertThat(values.get(0), equalTo(List.of((6))));
+        }
+    }
+
+    public void testMultipleMatch() {
+        var query = """
+            FROM test
+            | WHERE content MATCH "fox" OR content MATCH "brown"
+            | KEEP id
+            | SORT id
+            """;
+
+        try (var resp = run(query)) {
+            assertThat(resp.columns().stream().map(ColumnInfoImpl::name).toList(), equalTo(List.of(("id"))));
+            assertThat(resp.columns().stream().map(ColumnInfoImpl::type).map(DataType::toString).toList(), equalTo(List.of(("INTEGER"))));
+            // values
+            List<List<Object>> values = getValuesList(resp);
+            assertThat(values.size(), equalTo(5));
+            assertThat(values.get(0), equalTo(List.of((1))));
+            assertThat(values.get(1), equalTo(List.of((2))));
+            assertThat(values.get(2), equalTo(List.of((3))));
+            assertThat(values.get(3), equalTo(List.of((4))));
+            assertThat(values.get(4), equalTo(List.of((6))));
+        }
+    }
+
+    public void testMultipleWhereMatch() {
+        var query = """
+            FROM test
+            | WHERE content MATCH "fox" OR content MATCH "brown"
+            | EVAL summary = CONCAT("document with id: ", to_str(id), "and content: ", content)
+            | SORT summary
+            | LIMIT 4
+            | WHERE content MATCH "brown fox"
+            | KEEP id
+            """;
+
+        // TODO: this should not raise an error;
+        var error = expectThrows(ElasticsearchException.class, () -> run(query));
+        assertThat(error.getMessage(), containsString("Unsupported expression [content MATCH \"brown fox\"]"));
+    }
+
+    public void testNotWhereMatch() {
+        var query = """
+            FROM test
+            | WHERE NOT (content MATCH "brown fox")
+            | KEEP id
+            | SORT id
+            """;
+
+        try (var resp = run(query)) {
+            assertThat(resp.columns().stream().map(ColumnInfoImpl::name).toList(), equalTo(List.of(("id"))));
+            assertThat(resp.columns().stream().map(ColumnInfoImpl::type).map(DataType::toString).toList(), equalTo(List.of(("INTEGER"))));
+            // values
+            List<List<Object>> values = getValuesList(resp);
+            assertThat(values.size(), equalTo(1));
+            assertThat(values.get(0), equalTo(List.of((5))));
+        }
+    }
+
+    public void testNonExistingColumn() {
+        var query = """
+            FROM test
+            | WHERE something MATCH "fox"
+            """;
+
+        var error = expectThrows(VerificationException.class, () -> run(query));
+        assertThat(error.getMessage(), containsString("Unknown column [something]"));
```

Comment:
This doesn't feel like and IT on the pushdown - more like on `MATCH` itself. I might rename the class.

If you want to test the pushdown I might try using the `profile` option on the request and looking for the LuceneSourceOperator's status. It should have a list of queries it run and *that* should tell you that you pushed down.

I think you could do all of this with an `ESRestTestCase` if you wanted which gives you the advantage of being able to run it against many versions later.

For some of these failures I think I'd prefer them in the yaml test cases. They run against multiple versions by default and they have to be pretty simple so it can be easier to read them. Just because you know they can't surprise you.

Reply:
I would, also, be ok with the tests in VerifierTests (the VerificationException ones).